src/precompiles/mod.rs

```diff
@@ -39,94 +49,94 @@ impl<const B: bool> PrecompilesProcessor for DefaultPrecompilesProcessor<B> {
         memory: &mut M,
     ) -> Option<(Vec<MemoryQuery>, Vec<MemoryQuery>, PrecompileCyclesWitness)> {
         let address_low = u16::from_le_bytes([query.address.0[19], query.address.0[18]]);
-        match address_low {
-            KECCAK256_ROUND_FUNCTION_PRECOMPILE_ADDRESS => {
-                // pure function call, non-revertable
-                if B {
-                    let (reads, writes, round_witness) =
-                        keccak256::keccak256_rounds_function::<M, B>(
+        if let Ok(precompile_address) = PrecompileAddress::try_from(address_low) {
```

Comment:
Nit: IMHO we need to keep the depth of hte function smaller, i.e. rewrite to something like:

```
let Ok(x) = PrecompileAddress::try_from(address_low) else {
 return None;
} 

match x {
 ...
 }
```